src/v/archival/ntp_archiver_service.cc

```diff
@@ -685,18 +685,19 @@ ntp_archiver::schedule_uploads(model::offset last_stable_offset) {
 
 ss::future<std::vector<ntp_archiver::scheduled_upload>>
 ntp_archiver::schedule_uploads(std::vector<upload_context> loop_contexts) {
-    std::vector<scheduled_upload> scheduled_uploads;
-    auto uploads_remaining = _concurrency;
+    auto scheduled_uploads
+      = ss::make_lw_shared<std::vector<scheduled_upload>>();
+    auto uploads_remaining = ss::make_lw_shared(_concurrency);
     for (auto& ctx : loop_contexts) {
-        if (uploads_remaining <= 0) {
+        if (*uploads_remaining <= 0) {
```

Comment:
I was under the impression that as long as the caller `co_awaits` this method, these would remain in scope for the duration of this call. Is that not the case? Under what circumstances do these get destructed before returning (without making them shared)?